cmd/ctr/run.go

```diff
@@ -227,11 +232,22 @@ var runCommand = cli.Command{
 			return err
 		}
 		if len(spec) == 0 {
-			if spec, err = newSpec(context, &imageConfig.Config, ref); err != nil {
+			if spec, err = newContainerSpec(context, &imageConfig.Config, ref); err != nil {
 				return err
 			}
 		}
-		create, err := newCreateRequest(context, id, tmpDir, checkpoint, mounts, spec)
+
+		createContainer, err := newCreateContainerRequest(context, id, id, spec)
+		if err != nil {
+			return err
+		}
+
+		_, err = containers.Create(ctx, createContainer)
```

Comment:
need a matchin `containers.Delete()` otherwise the next `run` fails with `contianer <xxx> already exists`